pkg/sources/asdf/source.go

```diff
@@ -74,3 +76,38 @@ func New(cfg config.Config, out output.Output) (*Asdf, error) {
 func (a *Asdf) Name() string {
 	return "asdf"
 }
+
+func (a *Asdf) UpdateAll() (string, error) {
+	return a.execute("plugin", "update", "--all")
+}
+
+func (a *Asdf) InstallPlugin(plugin string) (string, error) {
+	output, err := a.execute("plugin", "list")
+	if err != nil {
+		return output, err
+	}
+
+	// We do not need to add the plugin first
+	// Maybe this should be a bit more safer than a simple contains of the whole output string
+	if strings.Contains(output, plugin) {
+		return output, err
+	}
+
+	return a.execute("plugin", "add", plugin)
+}
+
+func (a *Asdf) Install(args ...string) (string, error) {
+	output, err := a.UpdateAll()
+	if err != nil {
+		return output, err
+	}
+
+	output, err = a.InstallPlugin(args[0])
+	if err != nil {
+		return output, err
+	}
+
+	// This works but I'm not able to create a dynamic slice i.e. I tried:
+	// return a.execute(append([]string{"install"}, args...))
+	return a.execute("install", "golang", "1.19.2")
```

Comment:
How can  this work in Go?
`install` is stripped by the cobra command so we need to add the `install` again to the args.